CellProfiler_pipelines/README.md

```diff
@@ -0,0 +1,107 @@
+The goal of having multiple pipelines using various methods will allow for benchmarking for the "best" method later on. 
+
+## Illumination Correction (IC) Pipeline
+
+In the [NF1_illum.cpproj](Pipelines/NF1_illum.cpproj) CP project, I perform illumination correction on each channel (DAPI, GFP, RFP) of the NF1 data. 
+There are three sets of `CorrectIlluminationCalculate` and `CorrectIlluminationApply` modules, where each set corrects a different channel and each contains different set of parameters as I have found using the same parameters for each channel does not always work.
+
+For more information regarding these modules, you can reference my [Illumination Correction prototyping README](https://github.com/WayScience/CellProfiler_Prototyping/tree/main/1.illumination_correction) within the [CellProfiler_Prototyping repo](https://github.com/WayScience/CellProfiler_Prototyping). 
+This README describes each of the parameters within these modules and provides further references.
+
+Since the NF1 pilot data is small, this IC pipeline saves all of the images to use in the next pipeline.
+For much larger datasets, it is standard to save the illumination correction function (which is much smaller and saves storage) and use that to perform IC in the next pipeline.
+
+## Analysis Pipeline
+
+In the [NF1_analysis.cpproj](Pipelines/NF1_analysis.cpproj) CP project, I perform segmentation and feature extraction.
+
+### Segmentation
+
+To perform segmentation in CP, the modules that were used were:
+
+- `IdentifyPrimaryObjects` Identify nuclei from the DAPI channel images. This creates a group for an object called `OrigNuclei`, which includes nuclei that are a part of cells that touch an edge of the image.
+
+- `IdentifySecondaryObjects` Identify whole cells using the nuclei from the previous module as a base from the RFP channel (stained for actin) images. 
+These whole cells make up the object group called `Cells`.
+Any cells that are touching at least one edge of the image will be removed along with its respective nucleus. 
+This creates a new object group call `Nuclei`.
+
+- `IdentifyTeritaryObjects` Identify cytoplasm by subtracting out the "smaller identfied objects" (`OrigNuclei`) from the "larger identified objects" (`Cells`). 
+`OrigNuclei` has to be used in this module over `Nuclei` due to an error downstream in the pipeline when using the latter object where it states "Cytoplasm is not in a 1:1 relationship with other objects".
+This means that the number of cytoplasm is not the same as the other objects, which would be wrong because there should be the same amount of objects (e.g the number of cells should match the number of cytoplasm and nuclei).
```

Comment:
This to me seems like a red flag... Have you seen this issue discussed somewhere else?

Reply:
I agree. I found no forum post regarding this specific issue.

@gwaybio Do we make a forum post or issue on Github regarding this? Or just wait to see if there is an impact when testing cytosnake?

Reply:
Does David's answer here help?

https://forum.image.sc/t/exporttodatabase-error-an-identified-primary-object-not-in-1-1-relationship-with-the-others-preventing-export/56286

> For this particular pipeline, you’ll probably want to use “One table per object type” mode in ExportToDatabase. That should get it working.

Reply:
I saw that forum post and it doesn't help us since their issue is that they have multiple primary objects modules. We have a tertiary object that when using the filtered out Nuclei from the `IdentifySecondaryObjects` module as the smaller objects, it says that that Cytoplasm is not 1:1 with other objects.

@gwaybio I just tried changing that mode and it gave a warning saying that it could not be used in CellProfiler analyst fully. Is this "one table per object type" what we want to fit what cytosnake expects?

Reply:
gotcha, sounds like your way ahead of me :)

> Is this "one table per object type" what we want to fit what cytosnake expects?

I don't know what that option does, so maybe? Cytosnake expects a well-formuated SQLite

Reply:
I think you describe the SQLite file that I should be getting is a database with a table for Nuclei, a table for Cytoplasm, etc.

Based on this and the parameter description, that parameter seems to fit with your description. I will go with that when I rerun.